components/SalesHistoryTable/index.tsx

```diff
@@ -76,14 +123,20 @@ const SalesHistoryTable = ({ tableData, error }: Props): JSX.Element => {
 
   useEffect(() => {
     (async () => {
-      if (tableData.length) {
-        const chainAccounts = tableData.map(({ buyer }) => buyer);
-        const res = await getAvatars(chainAccounts);
-        setAvatars(res);
+      try {
+        if (renderedData.length) {
+          const chainAccounts = renderedData.map(({ buyer }) => buyer);
+          const res = await getAvatars(chainAccounts);
+          setIsLoading(false);
```

Comment:
Let's move `setIsLoading` to the end of the `try/catch` block so that if an error is thrown on line 129, `setIsLoading` is still invoked (otherwise, if an error is thrown, the table still stay loading).